rule.php

```diff
@@ -0,0 +1,97 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+use mod_quiz\local\access_rule_base;
+use mod_quiz\quiz_settings;
+use quizaccess_ai\ai_access_handler;
+
+/**
+ * A rule preventing access to quiz if AI is not available.
+ *
+ * @package   quizaccess_ai
+ * @copyright 2025 ISB Bayern
+ * @author    Thomas Schönlein
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class quizaccess_ai extends access_rule_base {
+    /**
+     * Factory: returns an instance of this rule if the quiz is suitable, null otherwise.
+     *
+     * @param quiz_settings $quizobj quiz settings object
+     * @param int $timenow current timestamp
+     * @param bool $canignoretimelimits whether time limits can be ignored (unused)
+     * @return access_rule_base|null the rule instance or null if not applicable
+     */
```

Comment:
if you are not writing anything specific to this subclass, just use `#[\Override]` instead of the phpdoc